app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,23 @@
+class MessagesController < ApplicationController
+
+  def index
+    @messages = Message.broadcast
+  end
+
+  def create
+    if params[:goal_id]
+      goal = Goal.find_by(id: params[:goal_id])
+    else
+      goal = Pledge.find_by(id: params[:pledge_id]).goal
```

Comment:
Do you ever get here without a params[:goal_id] ? The only route I can see that maps here is 
 goal_messages POST   /goals/:goal_id/messages(.:format)            messages#create